library/src/main/java/kwasm/format/text/MemoryType.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.format.text
+
+import kwasm.ast.Memory
+import kwasm.format.ParseException
+import kwasm.format.text.token.Keyword
+import kwasm.format.text.token.Paren
+import kwasm.format.text.token.Token
+
+/**
+ * From [the spec](https://webassembly.github.io/spec/core/text/types.html#memory-types):
+ * ```
+ *   memtype ::= lim:limits => lim
```

Comment:
This doesn't seem to match the code.  Does it need to include parens? or is the code off?

Reply:
Should have no parens

Reply:
I'm a bit confused here, when I've seen this in WASM code files its `(memory 1 34)`, so I should ignore the parens?

Reply:
Done.